benchmark-e2e/src/main/java/io/opentelemetry/e2ebenchmark/E2EAgentBenchmark.java

```diff
@@ -0,0 +1,122 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.e2ebenchmark;
+
+import java.util.*;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.testcontainers.containers.GenericContainer;
+import org.testcontainers.containers.Network;
+import org.testcontainers.containers.output.Slf4jLogConsumer;
+import org.testcontainers.containers.wait.strategy.Wait;
+import org.testcontainers.lifecycle.Startables;
+import org.testcontainers.utility.DockerImageName;
+import org.testcontainers.utility.MountableFile;
+
+public class E2EAgentBenchmark {
+  private static final Logger LOG = LoggerFactory.getLogger(E2EAgentBenchmark.class);
+  private List<GenericContainer<?>> containers;
+
+  @BeforeEach
+  void setUp() {
+    containers = new ArrayList<>();
+  }
+
+  @AfterEach
+  void tearDown() {
+    containers.forEach(GenericContainer::stop);
+  }
+
+  @Test
+  void run() throws Exception {
+    runBenchmark();
+  }
+
+  private void runBenchmark() throws Exception {
+    String agentPath = System.getProperty("io.opentelemetry.smoketest.agent.shadowJar.path");
+
+    // docker images
+    final DockerImageName WRK_IMAGE = DockerImageName.parse("quay.io/dim/wrk:stable");
+    final DockerImageName APP_IMAGE =
+        DockerImageName.parse(
+            "ghcr.io/open-telemetry/java-test-containers:smoke-springboot-jdk8-20201204.400701583");
```

Comment:
We could still have this reference an environment variable and default to this image name to allow benchmarking any image